packages/compartment-mapper/src/capture-lite.js

```diff
@@ -79,12 +85,160 @@ const captureCompartmentMap = (compartmentMap, sources) => {
 };
 
 /**
- * @param {ReadFn | ReadPowers} powers
+ * @type {LogFn}
+ */
+const noop = () => {};
+
+/**
+ * Factory for a function that loads compartments.
+ *
+ * @param {CompartmentMapDescriptor} compartmentMap Compartment map
+ * @param {Sources} sources Sources
+ * @param {LogOptions & PolicyOption & ForceLoadOption} [options]
+ * @returns {(linkedCompartments: Record<string, Compartment>, entryCompartment: Compartment, attenuatorsCompartment: Compartment) => Promise<void>}
+ */
+const makeLoadCompartments = (
+  compartmentMap,
+  sources,
+  { log = noop, policy, forceLoad = [] } = {},
+) => {
+  const {
+    entry: { module: entryModuleSpecifier },
+  } = compartmentMap;
+
+  /**
+   * Given {@link CompartmentDescriptor CompartmentDescriptors}, loads any which a) are present in the {@link forceLoad forceLoad array}, and b) have not yet been loaded.
+   *
+   * Will not load the "attenuators" `Compartment`, nor will it load any
+   * `Compartment` having a non-empty value in `sources` (since it is presumed it
+   * has already been loaded).
+   *
+   * @param {Record<string, Compartment>} compartments
+   * @returns {Promise<void>} Resolves when all appropriate compartments are loaded.
+   */
+  const forceLoadCompartments = async compartments => {
+    const compartmentsToLoad = forceLoad.reduce((acc, compartmentName) => {
+      // skip; should already be loaded
+      if (
+        compartmentName === ATTENUATORS_COMPARTMENT ||
+        compartmentName === compartmentMap.entry.compartment
+      ) {
+        return acc;
+      }
+
+      const compartmentDescriptor =
+        compartmentMap.compartments[compartmentName];
+
+      if (!compartmentDescriptor) {
+        throw new ReferenceError(
+          `Failed attempting to force-load unknown compartment ${q(compartmentName)}`,
+        );
+      }
+
+      const compartmentSources = sources[compartmentName];
+
+      if (keys(compartmentSources).length) {
+        log(
+          `Refusing to force-load Compartment ${q(compartmentName)}; already loaded`,
+        );
+        return acc;
+      }
+
+      const compartment = compartments[compartmentName];
+      if (!compartment) {
+        throw new ReferenceError(
+          `No compartment found for ${q(compartmentName)}`,
+        );
+      }
+      const compartmentOwnModuleDescriptor =
+        compartmentDescriptor.modules[compartmentDescriptor.name];
+
+      if (!compartmentOwnModuleDescriptor?.module) {
+        throw new Error(
```

Comment:
Is a `CompartmentDescriptor` _not having a `ModuleDescriptor` referencing itself_ considered malformed?